osu.Game/Screens/Multi/Multiplayer.cs

```diff
@@ -337,6 +338,12 @@ private void subScreenChanged(IScreen newScreen)
                     break;
             }
 
+            if (lastScreen is IOsuScreen lastOsuScreen)
```

Comment:
not hugely fond of this whole flow, but seems like about the best we can do with the current structure of multiplayer.